rust/README.md

```diff
@@ -0,0 +1,135 @@
+# Practical Networked Applications in Rust
+
+A training course on practical software construction in Rust.
+
+The final project in this course is a networked, persistent [key-value store]
+with multithreading, asynchronous I/O (via [tokio] and [tower]), a
+high-performance network protocol ([GRPC] via [prost] and [tower-grpc]),
+backed by a simple [LSM-tree] that the student will write themselves.
+
+Subjects covered include:
+
+- Structuring and maintaining Rust programs
+- Fun and foolproof parallel programming
+- Asyncronous programming with [futures] and [tokio]
+- High-performance networking with [GRPC] and [prost]
```

Comment:
do we need to use tower-grpc or grpc-rs?

Reply:
Either is fine with me. I put tower-grpc there since that's what we think we're heading to. Since it sounds like you prefer grpc-rs I'll use that and take out mentions of tower.

Reply:
no, I prefer pure Rust tower-grpc :-)